src/Database/Type/DateTimeType.php

```diff
@@ -157,7 +157,7 @@ public function marshal($value)
             if ($value === '' || $value === null || $value === false || $value === true) {
                 return null;
             }
-            if (is_numeric($value)) {
+            if (ctype_digit($value)) {
```

Comment:
Generally yes but not for all time
I worried about especial usage
`$calculate=...`
`$this->Models->newEntity(['data`=>$calculate])`